frontend/packages/kubevirt-plugin/src/models/kvReferenceForModel.ts

```diff
@@ -0,0 +1,37 @@
+import * as _ from 'lodash';
+import { modelFor, referenceForGroupVersionKind } from '@console/internal/module/k8s';
+import { K8sKind } from '@console/internal/module/k8s/types';
+
+const VERSIONS = ['v1', 'v1alpha3', 'v1beta1', 'v1alpha1'];
+
+export const kvReferenceForModel = (model: K8sKind): string => {
+  const availableVersion = _.uniq([model.apiVersion, ...VERSIONS]).find(
+    (v) => !!modelFor(referenceForGroupVersionKind(model.apiGroup)(v)(model.kind)),
+  );
+
+  if (availableVersion) {
+    return referenceForGroupVersionKind(model.apiGroup)(availableVersion)(model.kind);
+  }
+
+  return model.kind;
+};
+
+export const getKubevirtAvailableModel = (model: K8sKind): K8sKind => {
+  const availableVersion = _.uniq([model.apiVersion, ...VERSIONS]).find(
+    (v) => !!modelFor(referenceForGroupVersionKind(model.apiGroup)(v)(model.kind)),
+  );
+
+  if (availableVersion) {
+    return modelFor(referenceForGroupVersionKind(model.apiGroup)(availableVersion)(model.kind));
+  }
+
+  return model;
+};
+
+export const getKubevirtModelAvailableAPIVersion = (model: K8sKind): string => {
+  const availableVersion = _.uniq([model.apiVersion, ...VERSIONS]).find(
```

Comment:
Nit: you could extract the `availableVersion` computation into separate function to cut down code duplicity.

```ts
const getKubevirtAvailableModelVersion = (model: K8sKind): string =>
  _.uniq([model.apiVersion, ...VERSIONS]).find(
    (v) => !!modelFor(referenceForGroupVersionKind(model.apiGroup)(v)(model.kind)),
  );
```

and then replace `availableVersion` references with `getKubevirtAvailableModelVersion(model)` calls.